core/domain/prod_validation_jobs_one_off.py

```diff
@@ -5038,6 +5039,143 @@ def _get_custom_validation_functions(cls):
+            config_domain.WHITELISTED_EXPLORATION_IDS_FOR_PLAYTHROUGHS.value)
+
+        if item.exp_id not in whitelisted_exp_ids_for_playthroughs:
+            cls.errors['exploration id check'].append(
+                'Entity id %s: recorded in exploration_id:%s which '
+                'has not been curated for recording.' % (
+                    item.id, item.exp_id)
+            )
+
+    @classmethod
+    def _validate_reference(cls, item):
+        """Validate the playthrough reference relations.
+
+        Args:
+            item: ndb.Model. PlaythroughModel to validate.
+        """
+        exp_issues_tuples = cls.external_instance_details['exp_issues']
+        for _, _, exp_issues in exp_issues_tuples:
+            # The case for missing ExplorationIssues external model is
+            # ignored here since errors for missing email external model
+            # are already checked and stored in
+            # _validate_external_id_relationships function.
+            exp_id = item.exp_id
+            exp_version = item.exp_version
+
+            issues = []
+            for issue_index, issue in enumerate(exp_issues.unresolved_issues):
+                issue_type = issue['issue_type']
+                if (
+                        item.id in issue['playthrough_ids']
+                        and issue_type == item.issue_type):
+                    issue_customization_args = issue['issue_customization_args']
+                    identifying_arg = (
+                        stats_models.ISSUE_TYPE_KEYNAME_MAPPING[issue_type])
+                    if (
+                            issue_customization_args[identifying_arg] ==
+                            item.issue_customization_args[identifying_arg]):
+                        issues.append((issue_index, issue))
+
+            if len(issues) == 0:
+                cls.errors['reference check'].append(
+                    'Entity id %s: not referenced by any issue of the'
+                    ' corresponding exploration (id=%s, version=%s).' % (
+                        item.id, exp_id, exp_version)
+                )
+            elif len(issues) > 1:
+                issue_indices = [index for index, _ in issues]
+                cls.errors['reference check'].append(
+                    'Entity id %s: referenced by more than one issues of the '
+                    'corresponding exploration (id=%s, version=%s), '
+                    'issue indices: %s.' % (
+                        item.id, exp_id, exp_version, issue_indices)
+                )
+            else:
+                issue_index, issue = issues[0]
+                id_indices = []
+                reference_count = 0
+                for id_index, playthrough_id in enumerate(
+                        issue['playthrough_ids']):
+                    if playthrough_id == item.id:
+                        reference_count += 1
+                        id_indices.append(id_index)
+                if reference_count > 1:
```

Comment:
Looks amazing, thanks! Just one minor suggestion remaining: we no longer need `reference_count` since we now keep track of `id_indices`.

```suggestion
                for id_index, playthrough_id in enumerate(
                        issue['playthrough_ids']):
                    if playthrough_id == item.id:
                        id_indices.append(id_index)
                if len(id_indices) > 1:
```

Reply:
Sorry for the late reply - Done!